src/IDNAConverterTrait.php

```diff
@@ -125,7 +129,11 @@ private function idnToAscii(string $domain): string
             throw new UnexpectedValueException(sprintf('The Intl extension is misconfigured for %s, please correct this issue before proceeding.', PHP_OS));
         }
         // @codeCoverageIgnoreEnd
-
+    
+        if (property_exists($this, 'isTransitionalDifferent')) {
```

Comment:
I'd rather have the `$isTransitionalDifferent` property defined in the trait directly that having it being populated by the trait but not defined in it because of bad coupling.
the alternative is to create a protected `setIsTransitionalDifferent` method.